core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -529,20 +564,33 @@ class BrokerTopicMetrics(name: Option[String], configOpt: java.util.Optional[Kaf
   }
 }
 
-class BrokerTopicAggregatedMetric() {
+class BrokerTopicAggregatedMetric(allTopics: Boolean = false) {
```

Comment:
This class is specific to aggregated metrics on a topic. We are hacking it to provide broker level metrics. Maybe a BrokerAggregatedMetric is required?